core/app/models/spree/order/store_credit.rb

```diff
@@ -1,25 +1,13 @@
 module Spree
   class Order < Spree::Base
     module StoreCredit
-      def add_store_credit_payments
-        payments.store_credits.where(state: :checkout).map(&:invalidate!)
+      def add_store_credit_payments(amount = nil)
+        ActiveSupport::Deprecation.warn(<<-DEPRECATION, caller)
+          Spree::Order#add_store_credit_payments method is deprecated and will be removed in Spree 4.0.
+          Please use Spree::Checkout::AddStoreCredit service to remove item from cart.
```

Comment:
`Please use Spree::Checkout::AddStoreCredit service to add Store Credit to Order`